packaging/language/maven_artifact.py

```diff
@@ -208,13 +214,10 @@ def _request(self, url, failmsg, f):
                 "User-Agent": self.user_agent,
                 "Authorization": "Basic " + base64.b64encode(self.username + ":" + self.password)
             }
-        req = Request(url, None, headers)
-        try:
-            response = urlopen(req)
-        except HTTPError, e:
-            raise ValueError(failmsg + " because of " + str(e) + "for URL " + url)
-        except URLError, e:
-            raise ValueError(failmsg + " because of " + str(e) + "for URL " + url)
+
+        response, info = fetch_url(self.module, url, headers=headers)
```

Comment:
there's not currently a way to specify a specific file for an additional CA.  You can drop files into the directories specified in https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/urls.py#L427  /etc/ansible is probably the right directory to use if you only wanted it to affect tasks run by ansible.

A pull request could be made to add a parameter to allow a custom CA directory if you think it's needed.

Regarding basic auth, yeah -- I missed that when I made the port.